### DISCUSSION TITLE
ORTS-MG version(s) based on .Net 6 do not run under Linux (Ubuntu-based Linux Mint)

### DISCUSSION THREAD

**Discussion by mgavioli** [perpetualKid/FreeTrainSimulator#127] [General]
Context: Linux Mint 21.1, wine 8.0.1 (latest stable).

I understand that running under Linux/wine is not a top priority; still, as the wiki itself approaches this topic, it seems worth raising the point.

The ORTS-MG versions requiring .Net 6 crash under my Linux (I tried 1.4.11).

The last usable version (albeit with some shortcomings, like almost all items under "Tools" not doing anything) for me is 1.4.8.

Installation details (referring to version 1.4.11):

- create new wine prefix:
`   $> env WINEPREFIX="<prefix_path>" winecfg`
- in `winecfg` set to Windows 10 emulation;
- add  `corefonts` and `dotnet6` extensions:
`   $> env WINEPREFIX="<prefix_path>" winetricks corefonts dotnet6`
- extract "OpenRails_MG_1.4.11_Program.zip" in "<prefix_path>/drive_c";
- run `$> env WINEPREFIX="<prefix_path>" wine "<prefix_path>/drive_c/Program/net6.0-windows/Menu.exe"`

result:

![ORTS-MG_pre_dxvk](https://github.com/perpetualKid/ORTS-MG/assets/1420743/327e74f7-969a-44f3-860b-c3be454bc93b)

It looks like the .Net 6 built-in into `winetricks` is not enough (or not the correct one?). Following the link in the error dialogue box leads to `https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-6.0.16-windows-x64-installer?cid=getdotnetcore` downloading `windowsdesktop-runtime-6.0.16-win-x64.exe`

So, I deleted the whole prefix and started again from scratch:

- create new wine prefix:
`   $> WINEPREFIX="<prefix_path>" winecfg`
- in `winecfg` set to Windows 10 emulation;
- add `corefonts` extension:
`   $> WINEPREFIX="<prefix_path>" winetricks corefonts`
- run the previously downloaded .Net 6 package:
`   &> env WINEPREFIX="<prefix_path>" wine "<wine_cache_path>/windowsdesktop-runtime-6.0.16-win-x64.exe"`
- extract "OpenRails_MG_1.4.11_Program.zip" in "<prefix_path>/drive_c";
- run `$> env WINEPREFIX="<prefix_path>" wine "<prefix_path>/drive_c/Program/net6.0-windows/Menu.exe"`

result:

![ORTS-MG_pre_dxvk_2](https://github.com/perpetualKid/ORTS-MG/assets/1420743/9f623f72-d5f5-4331-8e5e-fcc38ab0e2e9)

and the programme seems stuck here (waiting brings no change and attempting to close it brings up the system box `<Programme> does not respond. [Force closing] / [Wait]`.

- install DXVK (as per its own github page instructions) and re-run `Menu.exe`: same result as above.

Any suggestion is welcome! Thanks!

**Comment by perpetualKid** [comment]
I have no idea about that, sorry!
Any contribution is welcome but I personally wont be able to engage on that topic.

**Comment by mgavioli** [comment]
Thanks for the reply. I did not actually expected Linux issues to be actually supported, I mostly posted this issue in case someone else run into it and/or is able to find some fix or work-around.

It is worth noting, though, that moving to .Net 6 did not improve portability but, at least for me, prevented it: I'll stick to 1.4.8, thanks for the good work!

**Comment by perpetualKid** [comment]
I'll move this to discussions so it doesn't get lost easily, but afaik there is no work done until someone volunteers and contributes.
